planb/update_cluster.py

```diff
@@ -311,11 +311,30 @@ def create_instance(ec2: object, volume: dict, saved_instance: dict,
     logger.info(
         "Creating new instance with IP {}".format(params['PrivateIpAddress'])
     )
-    ec2.run_instances(**params)
+    response = ec2.run_instances(**params)
+    instance_id = response['Instances'][0]['InstanceId']
+    create_tags(ec2, volume['VolumeId'], {'planb:operation:new-instance-id': instance_id})
     set_state(ec2, volume, 'created')
 
 
-def configure_instance(ec2: object, volume: dict, options: dict):
+def pre_configure_instance(ec2: object, volume: dict, saved_instance: dict):
+    if 'PublicIpAddress' in saved_instance:
```

Comment:
I think we could check if we need to assign public IP in `create_instance()` and then set the state to either 'created' or some other state which will be this extra step before going to 'created'.

Not sure what would be the good state name.  How about 'public-ip-needed'?  Then this step function name would be `assign_public_ip()`, for example.